sklearn/decomposition/nmf.py

```diff
@@ -831,7 +838,7 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
 
 
 def non_negative_factorization(X, W=None, H=None, n_components=None,
-                               init='random', update_H=True, solver='cd',
+                               init=None, update_H=True, solver='cd',
```

Comment:
Why are we changing this default init?

Reply:
Sorry for the super late reply and thanks for pointing this out. I should have explain it better in the PR notes.

I made this changes because the documentation says [here](https://github.com/scikit-learn/scikit-learn/blob/c759c4b56ea2e41ac28d2d6e54d1d63483daa593/sklearn/decomposition/nmf.py#L890)
```
        Default: 'nndsvd' if n_components <= min(n_samples, n_features),
            otherwise random.
```
But in `init` is set to `'nndsvd'` or `'random'` only if `init is None`. See [here](https://github.com/scikit-learn/scikit-learn/blob/c759c4b56ea2e41ac28d2d6e54d1d63483daa593/sklearn/decomposition/nmf.py#L315)
```
    if init is None:
        if n_components <= min(n_samples, n_features):
            init = 'nndsvd'
        else:
            init = 'random'
```

Reply:
Then we need to update the documentation to reflect the true default, not make a backwards-incompatible change. You are welcome to update the documentation in a separate, focused PR.

Reply:
Got it. Let me revert that and create a new PR on that.